File: boss_search.py
#!/usr/bin/env python3
import requests
import json
import time
import re

# Boss直聘API搜索
def search_jobs():
    # 城市代码: 深圳=101280600
    city_code = "101280600"
    keyword = "AI产品经理"
    
    cookies = {
        'wt2': 'DDp6j-2BDzXOgfzp3ymWACDixK36O1Kz9qFFlbQ_JwF537PcbmRVHslQK1d-LtrG-aA4ywS-6oFEvL2kPdY3Asw~~',
        'wbg': '0',
        'zp_at': 'qWS5gA_HUGex3aMZM0OZQYa0sCvxkcwsu4quKSIm04w~',
        'bst': 'V2QtgvFOL53FtgXdJuzR8cLyyx6zvSxA~~|QtgvFOL53FtgXdJuzRQZKSuw7DvSxA~~'
    }
    
    headers = {
        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36',
        'Accept': 'application/json, text/plain, */*',
        'Accept-Language': 'zh-CN,zh;q=0.9',
        'X-Requested-With': 'XMLHttpRequest',
        'Origin': 'https://www.zhipin.com'
    }
    
    all_jobs = []
    
    # 尝试获取多页数据
    for page in range(1, 4):
        print(f"正在获取第 {page} 页...")
        
        params = {
            'scene': '1',
            'query': keyword,
            'city': city_code,
            'page': str(page),
            'pageSize': '30'
        }
        
        try:
            response = requests.get(
                'https://www.zhipin.com/wapi/zpgeek/search/joblist.json',
                headers=headers,
                cookies=cookies,
                params=params,
                timeout=30
            )
            
            print(f"状态码: {response.status_code}")
            data = response.json()
            print(f"响应: {json.dumps(data, ensure_ascii=False, indent=2)[:500]}")
            
            if data.get('code') == 0:
                job_list = data.get('zpData', {}).get('jobList', [])
                print(f"第 {page} 页获取到 {len(job_list)} 条职位")
                
                for job in job_list:
                    job_info = {
                        'job_id': job.get('encryptJobId', ''),
                        'title': job.get('jobName', ''),
                        'salary': job.get('salaryDesc', ''),
                        'company': job.get('brandName', ''),
                        'company_id': job.get('encryptBrandId', ''),
                        'location': job.get('cityName', ''),
                        'district': job.get('areaDistrict', ''),
                        'business_district': job.get('businessDistrict', ''),
                        'experience': job.get('jobExperience', ''),
                        'education': job.get('jobDegree', ''),
                        'skills': job.get('skills', []),
                        'welfare': job.get('welfareList', []),
                        'company_size': job.get('brandScaleName', ''),
                        'company_stage': job.get('brandStageName', ''),
                        'company_industry': job.get('brandIndustry', ''),
                        'job_type': job.get('jobType', ''),
                        'link': f"https://www.zhipin.com/job_detail/{job.get('encryptJobId', '')}.html"
                    }
                    all_jobs.append(job_info)
            else:
                print(f"API错误: {data.get('message', '未知错误')}")
                
        except Exception as e:
            print(f"请求出错: {e}")
        
        time.sleep(2)
    
    # 保存结果
    output = {
        'search_time': time.strftime('%Y-%m-%d %H:%M:%S'),
        'keyword': keyword,
        'city': '深圳',
        'city_code': city_code,
        'total_count': len(all_jobs),
        'jobs': all_jobs
    }
    
    with open('/root/.openclaw/workspace/job_search_shenzhen.json', 'w', encoding='utf-8') as f:
        json.dump(output, f, ensure_ascii=False, indent=2)
    
    print(f"\n共获取 {len(all_jobs)} 条职位信息")
    print("结果已保存到 job_search_shenzhen.json")

if __name__ == '__main__':
    search_jobs()

File: boss_search.js
const { chromium } = require('playwright');
const fs = require('fs');

(async () => {
  const browser = await chromium.launch({ 
    headless: true,
    args: ['--no-sandbox', '--disable-setuid-sandbox']
  });
  
  const context = await browser.newContext({
    userAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/121.0.0.0 Safari/537.36',
    viewport: { width: 1920, height: 1080 }
  });
  
  // 设置Cookie
  const cookies = [
    { name: 'wt2', value: 'DDp6j-2BDzXOgfzp3ymWACDixK36O1Kz9qFFlbQ_JwF537PcbmRVHslQK1d-LtrG-aA4ywS-6oFEvL2kPdY3Asw~~', domain: '.zhipin.com', path: '/' },
    { name: 'wbg', value: '0', domain: '.zhipin.com', path: '/' },
    { name: 'zp_at', value: 'qWS5gA_HUGex3aMZM0OZQYa0sCvxkcwsu4quKSIm04w~', domain: '.zhipin.com', path: '/' },
    { name: 'bst', value: 'V2QtgvFOL53FtgXdJuzR8cLyyx6zvSxA~~|QtgvFOL53FtgXdJuzRQZKSuw7DvSxA~~', domain: '.zhipin.com', path: '/' }
  ];
  await context.addCookies(cookies);
  
  const page = await context.newPage();
  
  // 先访问首页
  console.log('访问Boss直聘首页...');
  await page.goto('https://www.zhipin.com', { waitUntil: 'domcontentloaded', timeout: 60000 });
  await page.waitForTimeout(3000);
  
  // 访问搜索页面
  const searchUrl = 'https://www.zhipin.com/web/geek/job?query=AI产品经理&city=101280600';
  console.log('访问搜索页面:', searchUrl);
  await page.goto(searchUrl, { waitUntil: 'networkidle', timeout: 60000 });
  await page.waitForTimeout(5000);
  
  // 截图查看页面状态
  await page.screenshot({ path: '/root/.openclaw/workspace/boss_search.png', fullPage: true });
  console.log('已截图保存到 boss_search.png');
  
  // 获取页面HTML内容
  const html = await page.content();
  fs.writeFileSync('/root/.openclaw/workspace/boss_page.html', html);
  console.log('页面HTML已保存');
  
  // 提取职位数据 - 尝试多种选择器
  const jobs = await page.evaluate(() => {
    const results = [];
    
    // 尝试多种可能的选择器
    const selectors = [
      '.job-card-wrapper',
      '.job-list .job-card',
      '[data-v-] .job-card',
      '.search-job-result .job-card',
      '.job-list-item'
    ];
    
    let jobCards = [];
    for (const selector of selectors) {
      jobCards = document.querySelectorAll(selector);
      if (jobCards.length > 0) {
        console.log('找到选择器:', selector, '数量:', jobCards.length);
        break;
      }
    }
    
    jobCards.forEach((card, index) => {
      try {
        const title = card.querySelector('.job-name, .name, .job-title, h3')?.textContent?.trim() || '';
        const salary = card.querySelector('.salary, .job-salary')?.textContent?.trim() || '';
        const company = card.querySelector('.company-name, .company, .comp-name')?.textContent?.trim() || '';
        const location = card.querySelector('.job-area, .area, .location')?.textContent?.trim() || '';
        const tags = Array.from(card.querySelectorAll('.tag-list li, .tag, .job-tags span')).map(li => li.textContent.trim());
        const link = card.querySelector('a')?.href || '';
        
        if (title) {
          results.push({
            title,
            salary,
            company,
            location,
            tags,
            link,
            index
          });
        }
      } catch (e) {
        console.error('解析职位卡片出错:', e);
      }
    });
    
    return results;
  });
  
  console.log('找到职位数量:', jobs.length);
  
  // 如果找到职位，滚动加载更多
  if (jobs.length > 0) {
    for (let i = 0; i < 3; i++) {
      await page.evaluate(() => window.scrollTo(0, document.body.scrollHeight));
      await page.waitForTimeout(3000);
    }
    
    // 再次提取所有职位
    const allJobs = await page.evaluate(() => {
      const results = [];
      const jobCards = document.querySelectorAll('.job-card-wrapper, .job-list .job-card, .job-list-item');
      
      jobCards.forEach(card => {
        try {
          const title = card.querySelector('.job-name, .name, .job-title, h3')?.textContent?.trim() || '';
          const salary = card.querySelector('.salary, .job-salary')?.textContent?.trim() || '';
          const company = card.querySelector('.company-name, .company, .comp-name')?.textContent?.trim() || '';
          const location = card.querySelector('.job-area, .area, .location')?.textContent?.trim() || '';
          const experience = card.querySelector('.tag-list li:nth-child(1), .experience')?.textContent?.trim() || '';
          const education = card.querySelector('.tag-list li:nth-child(2), .education')?.textContent?.trim() || '';
          const tags = Array.from(card.querySelectorAll('.tag-list li, .tag, .job-tags span')).map(li => li.textContent.trim());
          const link = card.querySelector('a')?.href || '';
          
          if (title) {
            results.push({
              title,
              salary,
              company,
              location,
              experience,
              education,
              tags,
              link
            });
          }
        } catch (e) {}
      });
      
      return results;
    });
    
    console.log('滚动后职位数量:', allJobs.length);
    
    // 保存结果
    const output = {
      search_time: new Date().toISOString(),
      keyword: 'AI产品经理',
      city: '深圳',
      city_code: '101280600',
      total_count: allJobs.length,
      jobs: allJobs
    };
    
    fs.writeFileSync('/root/.openclaw/workspace/job_search_shenzhen.json', JSON.stringify(output, null, 2));
    console.log('结果已保存到 job_search_shenzhen.json');
  } else {
    console.log('未找到职位数据，请检查页面结构');
    // 保存空结果
    const output = {
      search_time: new Date().toISOString(),
      keyword: 'AI产品经理',
      city: '深圳',
      city_code: '101280600',
      total_count: 0,
      jobs: [],
      error: '未找到职位数据'
    };
    fs.writeFileSync('/root/.openclaw/workspace/job_search_shenzhen.json', JSON.stringify(output, null, 2));
  }
  
  await browser.close();
})();
